ui/loader.c

```diff
@@ -0,0 +1,185 @@
+    return 0;
+}
+#endif
+
+/* Absolute path to the path of the current executable, gets filled in by `get_exe_path()` */
+char * exe_dir = NULL;
+static void * load_library(const char * rel_path) {
+    char path[2*PATH_MAX + 1];
+    snprintf(path, sizeof(path)/sizeof(char), "%s%s%s", exe_dir, PATHSEPSTRING, rel_path);
+
+    void * handle = NULL;
+#if defined(_OS_WINDOWS_)
+    wchar_t * wpath = NULL;
+    if (!utf8_to_wchar(path, &wpath)) {
+        fprintf(stderr, "ERROR: Unable to convert path %s to wide string!\n", path);
+        exit(1);
+    }
+    handle = (void *)LoadLibraryExW(wpath, NULL, LOAD_WITH_ALTERED_SEARCH_PATH);
+#else
+    handle = dlopen(path, RTLD_NOW | RTLD_GLOBAL);
+#endif
+
+    if (handle == NULL) {
+        fprintf(stderr, "ERROR: Unable to load dependent library %s\n", path);
+        exit(1);
+    }
+    return handle;
+}
+
+static void get_exe_path() {
+#if defined(_OS_WINDOWS_)
+    // On Windows, we use GetModuleFileName()
+    wchar_t julia_path[PATH_MAX];
+    if (!GetModuleFileName(NULL, julia_path, PATH_MAX)) {
+        fprintf(stderr, "ERROR: GetModuleFileName() failed with code %lu\n", GetLastError());
+        exit(1);
+    }
+    if (!wchar_to_utf8(julia_path, &exe_dir)) {
+        fprintf(stderr, "ERROR: Unable to convert julia path to UTF-8\n");
+        exit(1);
+    }
+#elif defined(_OS_DARWIN_)
+    // On MacOS, we use _NSGetExecutablePath(), followed by realpath()
+    char nonreal_exe_path[PATH_MAX + 1];
+    uint32_t exe_path_len = PATH_MAX;
+    int ret = _NSGetExecutablePath(nonreal_exe_path, &exe_path_len);
+    if (!ret) {
+        fprintf(stderr, "ERROR: _NSGetExecutablePath() returned %d\n", ret);
+        exit(1);
+    }
+
+    /* realpath(nonreal_exe_path) may be > PATH_MAX so double it to be on the safe side. */
+    exe_dir = (char *)malloc(2*PATH_MAX + 1);
+    if (realpath(nonreal_exe_path, exe_dir) == NULL) {
+        fprintf(stderr, "ERROR: realpath() failed with code %d\n", errno);
+        exit(1);
+    }
+#elif defined(_OS_LINUX_)
+    // On Linux, we read from /proc/self/exe
+    exe_dir = (char *)malloc(2*PATH_MAX + 1);
+    int num_bytes = readlink("/proc/self/exe", exe_dir, PATH_MAX);
+    if (num_bytes == -1) {
+        fprintf(stderr, "ERROR: readlink(/proc/self/exe) failed with code %d\n", errno);
+        exit(1);
+    }
+    exe_dir[num_bytes] = '\0';
+#elif defined(_OS_FREEBSD_)
+    // On FreeBSD, we use the KERN_PROC_PATHNAME sysctl:
+    exe_dir = (char *) malloc(2*PATH_MAX + 1);
+    int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+    int exe_dir_len = 2*PATH_MAX;
+    int ret = sysctl(mib, 4, exe_dir, &exe_dir_len, NULL, 0);
+    if (ret) {
+        fprintf(stderr, "ERROR: sysctl(KERN_PROC_PATHNAME) failed with code %d\n", ret);
+        exit(1);
+    }
+    exe_dir[exe_dir_len] = '\0';
+#endif
+
+    // Finally, convert to dirname
+    exe_dir = dirname(exe_dir);
+}
+
+#ifdef _OS_WINDOWS_
+int wmain(int argc, wchar_t *argv[], wchar_t *envp[])
```

Comment:
I believe that is `wmain` is available, you likely haven't linked this correctly. Most of those headers shouldn't be available either.